src/cluster/command/CommandContext.ts

```diff
@@ -103,9 +103,20 @@ export class CommandContext<TChannel extends KnownTextableChannel = KnownTextabl
 
         throw new Error('Cannot queryMember without a guild!');
     }
+    public async queryUser(options: SlimEntityFindQueryOptions): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(this: GuildCommandContext, options: Omit<SlimEntityFindQueryOptions, 'guild'>): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(options: SlimEntityPickQueryOptions<User>): Promise<ChoiceQueryResult<User>>;
+    public async queryUser(options: SlimEntityQueryOptions<User> | Omit<SlimEntityFindQueryOptions, 'guild'>): Promise<ChoiceQueryResult<User>> {
+        if ('choices' in options)
+            return await this.util.queryUser({ ...options, context: this.message, actors: this.author });
+
+        if ('guild' in options)
+            return await this.util.queryUser({ ...options, context: this.message, actors: this.author });
```

Comment:
could be merged into a single condition, right?

Reply:
sadly not. Logically yes, but typescript complains if you do
![image](https://user-images.githubusercontent.com/7736591/187523297-cc963211-49b6-488f-a414-5a86c2d6a87c.png)